# Introduction
This book has been inspired by recent convergence of two sciences, both of which are my life-long passions, both of which for the first time this year are becoming affordable to a an average person: genomics and cluster computers.
The field of genomics has exploded in the last few years beyond belief, the original human genome sequencing project, finished in year 2000, took 13 years and $3 billion to complete. Today, the cost of sequencing of the whole genome is approaching $800 (in bulk) and can be done in couple of hours.
The genome research has been concentrated around the prestigious institutions with generous grants that could afford access to newest sequencing technology. The positive outcome of the research sponsored by the public funds is that the results are also public and anyone can have access to genetic sequence information from the Web base databases and FTP sites. With a quick search you can get sequences of many organisms ranging from common bacteria, yeast, corn, wheat, fruit flies, mice, rats, extinct mammals, monkeys, apes, Neanderthal and many humans. Sequencing the next genome takes hours and there are thousands of them being sequenced now, as you read it.
For couple of hundreds of dollars you can determine presence of some interesting sequences using companies like 23andMe, best of all you can download the raw data of your test and start comparing it against other genomes or databases of genes immediately.
At the same time the medical field is learning about hundreds of thousands of proteins and trying to figure out which genetic sequences code for them. Doctors are discovering the genetic association of many diseases and individual drug interactions.
Each human genome is composed of 3.3 billion letters (base-pairs), comparing it against multiple other genomes requires some serious processing power. There are other organisms such as loblolly pine (Pinus taeda) that have 23 billion base pairs in their DNA, that is 7 time more than human!  Due to the sheer amount of data being generated every day there is a vast opportunity for new software tools and new applications of that knowledge.
The field of genomics is growing faster than any other technological advance in human history and few would argue that with potentially the biggest impact on our future lives since we learned how to use fire.

## Who is this book for?

This book is written for individuals who want to enter the field of genomic bioinformatics to develop their own software running on cluster computers. 
This book is NOT for people trying to run existing software packages that were designed for powerful, yet expensive workstations which usually rely on limited number of CPU cores (4~32) and on large amount of memory (RAM).
Finally, this book is for researchers without computer science background to consider these new computational solution. 
I would really enjoy if you, dear reader, contacted me to collaborate on developing new tools and solutions in some exciting new area.

## About the author

Uki D. Lucas has been a software developer, systems architect and development manager for over 18 years. In late 1980s he has spent countless hours over the microscope on Saturdays in extra-curricular biology classes. His interest in evolutionary biology and anthropology pre-date his computer career as he has studied Anthropology major in University of Maryland, Asian Division in Japan under Professor Janice Turner, Ph.D., to whom he will be forever thankful.
Uki has been developing Java systems, Android, has studied electronics in US Marines and automotive technology thereafter. Recently, he has also been working with Single Board Computers (SBC) for several years while working on systems that control large tractors and agricultural implements. Uki can be reached via email UkiDLucas@gmail.com, or via LinkedIn.
